main.js

```diff
@@ -30,31 +29,41 @@ function createWindow() {
   });
 
   mainWindowState.manage(mainWindow);
-  // and load the index.html of the app.
   mainWindow.loadURL("https://www.pandora.com");
 
-  mainWindow.on('close', function (event) {
-    // On macOS, most users are used to an application continuing to run
-    // in the background when the window is closed. This emulates the
-    // same behavior and allows closing the window to continue playing the
-    // radio.
-    if (process.platform === 'darwin') {
-      event.preventDefault();
-      mainWindow.hide();
-    }
-  });
+  // Only register Mac specific listeners if on Mac
+  if(process.platform === 'darwin') {
+    var wasForceQuit = false;
+
+    // Called before an app is quit. Should only be called when the app is actually quit ( https://electronjs.org/docs/api/app#event-before-quit )
+    app.on('before-quit', function(event) {
```

Comment:
Why is this in createWindow()? We should group all the app calls together at the bottom of this file. In this case, wasForceQuit would be in the outer scope and we could use let instead of var.

Reply:
Ah yes, was an oversight.

Reply:
Fixed in 6574de0ed630005878b0bacead7b366ebf371930